test/unit/core/selection.js

```diff
@@ -37,6 +37,42 @@ describe('Selection', function() {
     });
   });
 
+  describe('shadow root', function() {
```

Comment:
@jhchen am i ok to just disable these where `attachShadow` isn't supported? firefox tests will fail until i do as they're unusually far behind in implementing the shadow dom spec.